Mobile v2 docs update (#1188)
mattermost/mattermost-developer-documentation: site/content/contribute/more-info/mobile/developer-setup/_index.md

  - /contribute/mobile/developer-setup
---

The following instructions apply to the mobile apps for iOS and Android built in React Native.
Download the iOS version {{< newtabref href="https://apps.apple.com/us/app/mattermost/id1257222717" title="here" >}} and the Android version {{< newtabref href="https://play.google.com/store/apps/details?id=com.mattermost.rn" title="here" >}}.
Source code can be found at the {{< newtabref href="https://github.com/mattermost/mattermost-mobile" title="GitHub Mattermost Mobile app repository" >}}.

> If you run into any issues getting your environment set up, check the {{< newtabref href="https://docs.mattermost.com/deploy/mobile-troubleshoot.html" title="Troubleshooting" >}} section of the product docs for common solutions.

Reviewer: in this link I see
> First, confirm that your server URL has no typos and that it includes http:// or https:// according to the server deployment configuration.

There is no need to include the protocol.

> [I see a “Connecting…” bar that does not go away](https://docs.mattermost.com/deploy/mobile-troubleshoot.html#contents)

The bar no longer says connecting... @larkox can you provide the correct text being displayed?

> [None of these solve my problem!](https://docs.mattermost.com/deploy/mobile-troubleshoot.html#contents)

in this section there is a step to use `demo.mattermost.com` that server does not longer exists.

Author: These comments apply to a document outside of developers.mattermost.com.
@cwarnermm I think there needs to be a ticket posted to fix these up in the product docs.

Reviewer 2: Thanks, @neflyte. I created [a card](https://community.mattermost.com/boards/team/rcgiyftm7jyrxnma1osd8zswby/bbusacfknofye5jgtmrufo75mnc/v5hakifb3i7nw5emhjtbrdua7uy/cuwo9wu66j3fybgtm1izuqr9zno) to investigate further and update the docs where needed.

{{<note "iOS mobile app">}}
A macOS computer is required to build the Mattermost iOS mobile app.
{{</note>}}

## Environment setup

The following instructions apply to both iOS and Android mobile apps.
On macOS, we recommend using {{< newtabref href="https://brew.sh" title="Homebrew" >}} as a package manager.

### Install Cygwin (Windows only)

- {{< newtabref href="https://cygwin.com/faq/faq.html#faq.what.supported" title="Which versions of Windows does Cygwin support" >}}?
- Install Cygwin from {{< newtabref href="https://www.cygwin.com/" title="here" >}}.

> - Make sure to install the `make` and `patch` packages while installing Cygwin.

Reviewer: I'm not sure this is required anymore, have anyone tested the setup on Windows ?

Reviewer 2: @neflyte - Is testing the setup on Windows something you're open to helping with?

Author: Yes; I'll test the setup on Windows as far as I can. :)

- Using {{<newtabref title="NVM for Windows" href="https://github.com/coreybutler/nvm-windows">}}
{{</tab>}}

### Install Watchman

{{<newtabref href="https://facebook.github.io/watchman" title="Watchman">}} is a file watching program.
When a file changes, Watchman triggers an action, such as re-running a build command if a source file has changed.

The minimum required version of Watchman is 4.9.0.

{{<tabs "watchman" "watchman-mac,macOS;watchman-linux,Linux;watchman-windows,Windows" "watchman-mac">}}
{{<tab "watchman-mac" "display: block;">}}
To install Watchman using Homebrew, open a terminal and execute:

```sh
brew install watchman
```
{{</tab>}}
{{<tab "watchman-linux">}}
Download the latest package from {{< newtabref href="https://github.com/facebook/watchman/releases" title="here" >}}.
{{<note "Inotify limits">}}
Note that you need to increase your `inotify` limits for Watchman to work properly.
{{</note>}}
{{</tab>}}
{{<tab "watchman-windows">}}
Download the latest package from {{< newtabref href="https://github.com/facebook/watchman/releases" title="here" >}}.
{{</tab>}}

### Install `react-native-cli` tools

```sh

> npm -g install react-native-cli

Reviewer: not needed anymore

```sh
export ANDROID_HOME=$HOME/Library/Android/sdk
export PATH=$ANDROID_HOME/emulator:$ANDROID_HOME/platform-tools:$ANDROID_HOME/tools:$PATH
```

Then reload your bash configuration:

```sh
source ~/.bash_profile
```

{{<note>}}
Depending on the shell you're using, this might need to be put into a different file such as `~/.zshrc`. Adjust this accordingly.
{{</note>}}
{{</tab>}}
{{<tab "droid-linux">}}
On Linux the home folder is located under `/home/<username>` which results in a slightly different path:

```sh
export ANDROID_HOME=/home/<username>/Android/Sdk
export PATH=$ANDROID_HOME/platform-tools:$PATH
export PATH=$ANDROID_HOME/tools:$PATH
```

Then reload your configuration

```sh
source ~/.bash_profile
```

{{<note>}}
Depending on the shell you're using, this might need to be put into a different file such as `~/.zshrc`. Adjust this accordingly.
{{</note>}}
{{</tab>}}

### Install the SDKs and SDK tools

In the SDK Manager using Android Studio or the {{< newtabref href="https://developer.android.com/studio/command-line/sdkmanager.html" title="Android SDK command line tool" >}}, ensure the following are installed:

> - SDK Tools (you may have to click "Show Package Details" to expand packages):

Reviewer 2: ```suggestion
- SDK Tools (you may have to select **Show Package Details** to expand packages):
```

>   - Android SDK Build-Tools 29.0.2

Reviewer: this should be 31 not 29.0.2

  - Android Emulator
  - Android SDK Platform-Tools
  - Android SDK Tools
  - Google Play services
  - Intel x86 Emulator Accelerator (HAXM installer)
  - Support Repository
    - Android Support Repository
    - Google Repository

  ![image](sdk_tools.png)

> - SDK Platforms (you may have to click "Show Package Details" to expand packages)

Reviewer 2: ```suggestion
- SDK Platforms (you may have to select **Show Package Details** to expand packages)
```

>   - Android 7 (Nougat) or above
>     - Google APIs
>     - SDK Platform
>       - For Android Q or above > Android SDK Platform 29 or above
>     - Intel or Google Play Intel x86 Atom\_64 System Image
>   - Any other API version that you want to test

Reviewer: ```suggestion
  - Android 12  or above
    - Google APIs
    - SDK Platform
      - Android SDK Platform 31 or above
    - Intel or Google Play Intel x86 Atom\_64 System Image
  - Any other API version that you want to test
```

  ![image](sdk_platforms.png)

{{<note "Android 5/6 support">}}
We've dropped Android 5/6 Support since December 2018; you may still continue to use 1.14 for Android 5/6 devices. See the following GitHub issue for more information: {{<newtabref href="https://github.com/mattermost/mattermost-mobile/issues/2480" title="Android 5/6 support">}}

> {{</note>}}

Reviewer: I don't think this note is needed

## Obtain the source code

In order to develop and build the Mattermost mobile apps, you'll need to get a copy of the source code. Forking the `mattermost-mobile` repository will also make it easy to contribute your work back to the project in the future.

1. Fork the {{< newtabref href="https://github.com/mattermost/mattermost-mobile" title="mattermost-mobile" >}} repository on GitHub.

2. Clone your fork locally:

>    - Open a terminal

Reviewer 2: Please convert these bulleted items with a numbered list of sub-steps (a., b., c.,) so that users know to perform each step in the order prescribed.

Check the following things:

* Ensure you are running the latest version of `nvm` using the {{< newtabref href="https://github.com/nvm-sh/nvm#install--update-script" title="upgrade instructions" >}}.
* Ensure you have set your desired version of node in the file `~/.nvmrc`.  E.g.,
  ```sh
  echo v16.2.0 > ~/.nvmrc

>   ```

Reviewer: I'm not sure this is needed, but in case we want to keep it the version should be v18+

Also the section of `PhaseScriptExecution` can be removed
